cryptographic_fields/fields.py

```diff
@@ -25,30 +29,37 @@
 crypter = cryptography.fernet.MultiFernet(keys)
 
 
-def encrypt_str(s):
-    # be sure to encode the string to bytes
+def encrypt_str(s, salt):
+    s = u'{salt}:{string}'.format(
+        salt=os.urandom(salt).encode('base_64'), string=s)
+    # be sure to encode the string to bytes.
     return crypter.encrypt(s.encode('utf-8'))
 
 
-def decrypt_str(t):
+def decrypt_str(t, salt):
     # be sure to decode the bytes to a string
-    return crypter.decrypt(t.encode('utf-8')).decode('utf-8')
+    raw_text = crypter.decrypt(t.encode('utf-8')).decode('utf-8')
+    # remove the salt
+    return raw_text.split(':', 1)[1]
 
 
-def calc_encrypted_length(n):
+def calc_encrypted_length(n, salt):
     # calculates the characters necessary to hold an encrypted string of
     # n bytes
-    return len(encrypt_str('a' * n))
+    return len(encrypt_str('a' * n, salt))
 
 
 class EncryptedMixin(object):
     def __init__(self, *args, **kwargs):
+        self.salt_length = kwargs.pop('salt', 2)
+
         super(EncryptedMixin, self).__init__(*args, **kwargs)
         # set the max_length to be large enough to contain the encrypted value
         if not self.max_length:
             self.max_length = 10
         self.unencrypted_max_length = self.max_length
-        self.max_length = calc_encrypted_length(self.unencrypted_max_length)
+        self.max_length = calc_encrypted_length(
+            self.unencrypted_max_length, self.salt_length)
```

Comment:
@LucasRoesler this actually doesn't look quite right. You're passing in a salt_length (an number), but its being used in the ```calc_encrypted_length``` as the actual salt value.